### PR TITLE
Update docs for inlay hints

### PR DESCRIPTION
**Pull request by rpfaeffle**
Follow up of #7943. Updates the docs for inlay hints as they are incorrect (for Svelte).

Release Notes:

- N/A

### CODE REVIEW

File: docs/src/configuring_zed.md [hunk 1]
```diff
@@ -393,93 +393,14 @@ Inlay hints querying consists of two parts: editor (client) and LSP server.
...
+- [Go](https://docs.zed.dev/languages/go)
+- [Rust](https://docs.zed.dev/languages/rust)
+- [Svelte](https://docs.zed.dev/languages/svelte)
+- [Typescript](https://docs.zed.dev/languages/typescript)
 
-```json
-"lsp": {
-  "rust-analyzer": {
-    "initialization_options": {
-      "inlayHints": {
```
Comment (SomeoneToIgnore) [inline docs/src/configuring_zed.md:404]: This is getting lost due to https://github.com/zed-industries/zed/blob/main/docs/src/languages/rust.md having nothing related to inlay hints at all, let's rather move it there and link https://rust-analyzer.github.io/manual.html#inlay-hints too?

**Review by SomeoneToIgnore — COMMENTED** [review]
Thank you for noticing (and keeping our docs better!).
LGTM when Rust gets back its inlay hints' docs.

File: docs/src/configuring_zed.md (hunk 1 above)
Comment (rpfaeffle) [reply docs/src/configuring_zed.md:404]: I must have missed that. That sounds right to me. I'll update the docs.

File: docs/src/languages/rust.md [hunk 2]
```diff
@@ -3,7 +3,42 @@
 - Tree Sitter: [tree-sitter-rust](https://github.com/tree-sitter/tree-sitter-rust)
 - Language Server: [rust-analyzer](https://github.com/rust-lang/rust-analyzer)
 
-### Target directory
+## Inlay Hints
+
+Zed sets the following initialization options for inlay hints:
```
Comment (rpfaeffle) [inline docs/src/languages/rust.md:8]: Is this the default, or must this be configured by the user?

Comment (SomeoneToIgnore) [reply docs/src/languages/rust.md:8]: It's not, actually, just a configuration example.
There's no extra hint settings Zed sets for r-a.

**Review by SomeoneToIgnore — APPROVED** [review]
❤️

**Pull request merged**